src/parser/jobs/rdm/modules/ManaStackGauge.tsx

```diff
@@ -0,0 +1,214 @@
+
+interface ManaStackGaugeModifier {
+	manaStack: number
+}
+interface GaugeBreaker {
+	action: number,
+	timestamp: number,
+	manaStacksLost: number,
+}
+export const MINIMUM = 0
+export const MAXIMUM = 3
+
+export class ManaStackGauge extends CoreGauge {
+	static override handle = 'manaStackGauge'
+	static override title = t('rdm.manaStackGauge.title')`Mana Stack Gauge Usage`
+	static override debug = false
+
+	@dependency private suggestions!: Suggestions
+	@dependency private statistics!: Statistics
+	@dependency private gauge!: Gauge
+
+	private manaStackGauge = this.add(new CounterGauge({
+		graph: {
+			label: <Trans id="rdm.manaStackGauge.resource.manaStacks">Mana Stacks</Trans>,
+			color: JOBS.RED_MAGE.colour,
+		},
+		maximum: MAXIMUM,
+		minimum: MINIMUM,
+	}))
+	public gaugeModifiers = new Map<number, ManaStackGaugeModifier>([
+		[this.data.actions.ENCHANTED_REPRISE.id, {manaStack: 1}],
+		[this.data.actions.ENCHANTED_MOULINET.id, {manaStack: 1}],
+		[this.data.actions.ENCHANTED_RIPOSTE.id, {manaStack: 1}],
+		[this.data.actions.ENCHANTED_ZWERCHHAU.id, {manaStack: 1}],
+		[this.data.actions.ENCHANTED_REDOUBLEMENT.id, {manaStack: 1}],
+	])
+	public spenderModifiers = new Map<number, ManaStackGaugeModifier>([
+		[this.data.actions.VERHOLY.id, {manaStack: -1}],
+		[this.data.actions.VERFLARE.id, {manaStack: -1}],
+		[this.data.actions.SCORCH.id, {manaStack: -1}],
+		[this.data.actions.RESOLUTION.id, {manaStack: -1}],
+	])
+	public gaugeBreakers: number[] = [
+		this.data.actions.VERAERO.id,
+		this.data.actions.VERAERO_II.id,
+		this.data.actions.VERAERO_III.id,
+		this.data.actions.VERSTONE.id,
+		this.data.actions.VERTHUNDER.id,
+		this.data.actions.VERTHUNDER_II.id,
+		this.data.actions.VERTHUNDER_III.id,
+		this.data.actions.VERFIRE.id,
+		this.data.actions.JOLT.id,
+		this.data.actions.JOLT_II.id,
+		this.data.actions.IMPACT.id,
+	]
+	gaugeBreak: GaugeBreaker[] = []
+	//Any loss is a major severity
+	severity = {
+		1: SEVERITY.MAJOR,
+	}
+	manaStatistics = {
+		white: {
+			overcapLoss: 0,
+		},
+		black: {
+			overcapLoss: 0,
+		},
+	}
+
+	override initialise() {
+		super.initialise()
+
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(oneOf(Array.from(this.gaugeModifiers.keys()))),
+			this.onGaugeModifying
+		)
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(oneOf(this.gaugeBreakers)),
+			this.onGaugeModifying
+		)
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(oneOf(Array.from(this.spenderModifiers.keys()))),
+			this.onGaugeSpender
+		)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onGaugeModifying(event: Events['action']) {
+		const modifier = this.gaugeModifiers.get(event.action)
+
+		//Always check it we should break
+		if (this.onGaugeBreak(event)) {
+			//We won't modify the gauge if this skill broke it.
```

Comment:
This comment is a bit misleading. You are obviously resetting the gauge, it's just happening as part of the `if` statement.

Reply:
will rephrase it.